app/services/firestore.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  DocumentReference,
+  doc,
+  getDoc,
+  setDoc,
+} from 'firebase/firestore';
+import type { userDataSanitization } from '../schemas/userDataSanitizationSchema';
+import { db } from '../config/firebase';
+
+/**
+ * Asynchronously checks the existence of a Firestore document.
+ *
+ * @param {DocumentReference} docRef - Reference to the Firestore document.
+ * @returns {Promise<boolean>} True if the document exists, false otherwise or if an error occurs.
+ */
+export const doesUserDataExist = async (
+  userDocRef: DocumentReference
+): Promise<boolean> => {
+  try {
+    const documentSnapshot = await getDoc(userDocRef);
+    return documentSnapshot.exists();
+  } catch (error) {
+    console.error('Error checking user data existence.');
+    return false;
+  }
+};
+
+/**
+ * Stores user data in Firestore if the user does not already exist.
+ *
+ * @param {string} userId - The unique identifier for the user.
+ * @param {userDataSanitization} userData - The sanitized user data to be stored.
+ * @returns {Promise<void>} - A promise that resolves when the operation is complete.
+ */
+export const storeUserDataIfNew = async (
+  userId: string,
```

Comment:
This is a minor naming suggestion. Since we are using user's email as the identifier, would it be best to use `userEmail` instead of `userId`?

Reply:
This param is intentionally named `userId` to indicate we are passing in the user email as the user id (as it is not super common to use an email for a doc id):)

Reply:
Gotcha. would it be a good idea to clarify that in jsdoc? Ironically, i was abit confused when i first saw the `userId` param, even though i wrote the create user data logic initially and knew we use user email as identifier.